src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.cs

```diff
@@ -5356,6 +5356,72 @@ public object F()
             var f0 = compilation0.GetMember<MethodSymbol>("C.F");
             var f1 = compilation1.GetMember<MethodSymbol>("C.F");
 
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            diff1.EmitResult.Diagnostics.Verify();
+
+            diff1.VerifySynthesizedMembers(
+                "C: {<>c}",
+                "C.<>c: {<>9__0_0, <F>b__0_0, <<F>b__0_0>d}",
+                "C.<>c.<<F>b__0_0>d: {<>1__state, <>t__builder, <>4__this, <a>5__1, <>s__2, <>u__1, MoveNext, SetStateMachine}");
+        }
+
+        [ConditionalFact(AlwaysSkip = "")]
+        public void AsyncMethodWithNullableParameterAddingNullCheck()
+        {
+            var source0 = MarkedSource(@"
+using System;
+using System.Threading.Tasks;
+#nullable enable
+
+class C
+{
+    static T id<T>(T t) => t;
+    static Task<T> G<T>(Func<T> f) => Task.FromResult(f());
+
+    public async void F(string? x)
+    {
+        var <N:2>y = await G(<N:0>() => new { A = id(x) }</N:0>)</N:2>;
+        var <N:3>z = new Func<string>(<N:1>() => y.A</N:1>)</N:3>;
+    }
+}
+", options: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+            var source1 = MarkedSource(@"
+using System;
+using System.Threading.Tasks;
+#nullable enable
+
+class C
+{
+    static T id<T>(T t) => t;
+    static Task<T> G<T>(Func<T> f) => Task.FromResult(f());
+
+    public async void F(string? x)
+    {
+        if (x is null) throw new Exception();
+        var <N:2>y = await G(<N:0>() => new { A = id(x) }</N:0>)</N:2>;
+        var <N:3>z = new Func<string>(<N:1>() => y.A</N:1>)</N:3>;
+    }
+}
+", options: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+
+            var compilation0 = CreateCompilationWithMscorlib45(new[] { source0.Tree });
```

Comment:
>}); [](start = 84, length = 3)

The compilation doesn't produce debug information, hence the closure map is missing. Pass `options: ComSafeDebugDll`.
You can place breakpoint to `EncVariableSlotAllocator.TryGetPreviousClosure` to see that the `_closureMapOpt` is null.